pkg/topology/topology.go

```diff
@@ -39,6 +41,24 @@ const (
 	Balanced
 )
 
-// LinstorNodeKey refers to a node running the LINSTOR csi node service
-// and the linstor Satellite and is therefore capabile of hosting LINSTOR volumes.
-const LinstorNodeKey = "linbit.com/hostname"
+const (
+	// LinstorNodeKey refers to a node running the LINSTOR csi node service
+	// and the linstor Satellite and is therefore capable of hosting LINSTOR volumes.
+	LinstorNodeKey = "linbit.com/hostname"
+
+	// LinstorStoragePoolKeyPrefix is the prefix used when specifying the available storage
+	// pools on a node via CSI topology keys.
+	LinstorStoragePoolKeyPrefix = "linbit.com/sp"
+
+	// The value assigned to the storage pool label, given that the node has access to the storage pool.
+	LinstorStoragePoolValue = "true"
+)
+
+// Converts the storage pool name into a CSITopology compatible label.
+//
+// For example: "abcd-efg" will be converted to "linbit.com/sp-abcd-efg".
+// There is an upper limit on the length of these keys, so the storage pool name may be truncated to fit the maximum
+// label length.
+func ToStoragePoolLabel(storagePoolId string) string {
```

Comment:
what is this limit and where is it defined? Can we use that value here? And I'd prefer to have the overall length then be calculated based on the length of this and `LinstorStoragePoolKeyPrefix`. Or glue them together and do a `[:max]`.

Is it possible that we generate collisions? `maxCSI < maxLINSTORSPname + len(LinstorStoragePoolKeyPrefix)`.

Reply:
Yeah, its a bit of a mess unfortunately. The limit is described in the CSI Spec (Look for `message Topology {`): https://github.com/container-storage-interface/spec/blob/v1.0.0/spec.md

Its defined as a "prefix" which is everything before the `/`, max 63 characters, and the actual key, which again is only allowed to contain 63 characters. Also this: 
> `Keys MUST be case-insensitive`

So I'm not sure how to proceed. We could hash the LINSTOR name, which should keep us under the 63 character limit without fear of collisions.

Reply:
the LINSTOR name is max 48 chars, so we should always be on the safe side. just document it here. maybe do a check+panic for the impossible case?

Reply:
Panic should be fine in this case.